core/domain/exp_jobs_one_off_test.py

```diff
@@ -1246,6 +2032,40 @@ def test_for_validation_job(self):
 
         self.assertEqual(actual_output, expected_output)
 
+    def test_no_action_is_performed_for_deleted_exploration(self):
```

Comment:
I see this is a common test for all the jobs. Is there a possibility to extract some of the common part into a helper function. This I think can help enhance readability. I am ok with it as is. 

I think something along the lines of:
- A function that creates and deletes an exploration. 
- A function that takes a job id and runs the job, verifies no output is produced.

And then these 2 functions can be called across all these tests.

Reply:
I am not sure about this.

There are some issues which obstruct this idea:

- In some of the test, we need to make changes to exploration states between creation and deletion.
- In some of the test, instead of checking that no output is produced we need to check if no error is raised when a specific function call is made.

We can make changes according to these situations as well -- avoid the creation and deletion function. Create a helper function to run the job which accepts a flag whether to verify output or check error message. If the case is to check error message, extra parameters denoting the error and function call will also be passed to the function.

Does this sound good?

Reply:
Yeah, I thought so too.. I think what you say sounds good. That should help enhance readability I feel!

Reply:
Done, PTAL!

Reply:
Have to say, this refactor looks amazing! Much more readable, and very well written function :)

Reply:
Thanks a lot :smile: